### ISSUE TITLE
Why not mask unk_token when actor generation?

### ISSUE THREAD

**Issue by ZiyiLiubird** [OpenLLMAI/OpenRLHF#176]
https://github.com/OpenLLMAI/OpenRLHF/blob/df92b437a34ec832648a275949f493574dcded68/openrlhf/models/actor.py#L109

maybe

attention_mask = (sequences.ne(unk_token_id) & sequences.ne(pad_token_id)).to(dtype=torch.long)

**Comment by hijkzzz** [comment]
The model usually does not output UNK_TOKEN_ID

**Comment by ZiyiLiubird** [comment]
> The model usually does not output UNK_TOKEN_ID

thanks for your reply. If there are multiple eos_tokens in prompt, for example, multi-turn dialogues and concat them via

```python
    formatted_text = ""
    for idx, message in enumerate(messages):
        if message["role"] == "system":
            formatted_text += "<|system|>\n" + message["content"] + "\n"
        elif message["role"] == "user":
            formatted_text += "<|user|>\n" + message["content"] + "\n"
        elif message["role"] == "assistant":
            formatted_text += "<|assistant|>\n" + message["content"].strip() + eos_token + "\n"
      
    formatted_text += "<|assistant|>\n"
    return formatted_text
```
then whether it is compatible with this case?

**Comment by hijkzzz** [comment]
It seems unreasonable to use data in this format in RLHF

**Comment by ZiyiLiubird** [comment]
> It seems unreasonable to use data in this format in RLHF

For example, role-play via RLHF.
